File: Chess/src/main/java/side_client/ClientListenThread.java
package side_client;

import game.boards.Board;
import game.move.Move;
import game.pieces.PieceTypes;
import game.pieces.Team;
import game.player.Player;
import messages.Message;
import messages.MovementMessage;

import javax.swing.*;
import java.awt.*;
import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;

public class ClientListenThread extends Thread {

    Client client;

    public ClientListenThread(Client client) {
        this.client = client;
    }

    @Override
    public void run() {
        while (!this.client.socket.isClosed()) {

            try {

                Message msg = (Message) (this.client.sInput.readObject());
                switch (msg.type) {
                    case START:
                        Team serverChosenTeam = (Team) msg.content;
                        this.client.setTeam(serverChosenTeam);
                        break;
                    case PAIRING:
                        this.client.isPaired = true;
                        this.client.game.getMainMenu().getPlayBTN().setEnabled(true);
                        this.client.game.getMainMenu().getPlayBTN().setText("side_client.Start Game");
                        this.client.game.getMainMenu().getInfoLBL().setText("Matched. Click To side_client.Start Game");
                        break;
                    case MOVE:
                        MovementMessage movement = (MovementMessage) msg.content;
                        Board board = this.client.game.getChessBoard();
                        Player player = board.getCurrentPlayer();
                        Move move = new Move(board, board.getTile(movement.currentCoordinate), board.getTile(movement.destinationCoordinate));
                        player.makeMove(board, move);
                        this.client.game.getBoardPanel().updateBoardGUI(this.client.game.getChessBoard());
                        if (move.hasKilledPiece()) {
                            if (move.getKilledPiece().getType() == PieceTypes.KING) {
                                Team winnerTeam;
                                winnerTeam = (move.getKilledPiece().getTeam() == Team.BLACK) ? Team.WHITE : Team.BLACK;
                                JOptionPane.showMessageDialog(null, "Winner: " + winnerTeam.toString());
                                Message message = new Message(Message.MessageTypes.END);
                                message.content = null;
                                client.Send(message);
                                break;
                            }
                        }
                        board.changeCurrentPlayer();
                        this.client.game.getBottomGameMenu().getTurnLBL().setText("Your Turn");
                        this.client.game.getBottomGameMenu().getTurnLBL().setForeground(Color.GREEN);
                        break;
                    case CHECK:

                        Team checkStateTeam = (Team) msg.content;
                        JOptionPane.showMessageDialog(null, "Check state for team: " + checkStateTeam.toString());
                        break;

                    case LEAVE:
                        JOptionPane.showMessageDialog(null, "Enemy left. Returning to the Menu.");
                        this.client.game.getGameFrame().remove(this.client.game.getBoardPanel());
                        this.client.game.createMainMenu();
                }

            } catch (IOException ex) {
                Logger.getLogger(ClientListenThread.class.getName()).log(Level.SEVERE, null, ex);
            } catch (ClassNotFoundException ex) {
                System.out.println("Girilen class bulunamadı");
            }
        }
    }
}

File: Chess/src/main/java/side_client/ClientListenThread.kt
package side_client

import game.Move
import game.piece.PieceTypes
import game.piece.Team
import msg.Message
import msg.Message.MessageTypes
import msg.MovementMessage
import java.awt.Color
import java.io.IOException
import java.util.logging.Level
import java.util.logging.Logger
import javax.swing.JOptionPane

class ClientListenThread(var client: Client) : Thread() {
    override fun run() {
        while (!client.socket!!.isClosed) {
            try {
                val msg = client.sInput!!.readObject() as Message
                when (msg.type) {
                    MessageTypes.START -> {
                        val serverChosenTeam = msg.content as Team?
                        client.team = serverChosenTeam!!
                    }

                    MessageTypes.PAIRING -> {
                        client.isPaired = true
                        client.game.mainMenu.playBTN?.isEnabled = true
                        client.game.mainMenu.playBTN?.text = "Start Game"
                        client.game.mainMenu.infoLBL?.text = "Matched. Click To Start Game"
                    }

                    MessageTypes.MOVE -> {
                        val movement = msg.content as MovementMessage?
                        val board = client.game.chessBoard
                        val player = board.currentPlayer
                        val move = Move(
                            board, board.getTile(movement!!.currentCoordinate), board.getTile(
                                movement.destinationCoordinate
                            )
                        )
                        player.makeMove(board, move)
                        client.game.boardPanel.updateBoardGUI(client.game.chessBoard)
                        if (move.hasKilledPiece()) {
                            if (move.killedPiece.type === PieceTypes.KING) {
                                val winnerTeam: Team = if (move.killedPiece.team === Team.BLACK) Team.WHITE else Team.BLACK
                                JOptionPane.showMessageDialog(null, "Winner: $winnerTeam")
                                val message = Message(MessageTypes.END)
                                message.content = null
                                client.send(message)
                                break
                            }
                        }
                        board.changeCurrentPlayer()
                        client.game.bottomGameMenu.turnLBL.text = "Your Turn"
                        client.game.bottomGameMenu.turnLBL.foreground = Color.GREEN
                    }

                    MessageTypes.CHECK -> {
                        val checkStateTeam = msg.content as Team?
                        JOptionPane.showMessageDialog(null, "Check state for team: " + checkStateTeam.toString())
                    }

                    MessageTypes.LEAVE -> {
                        JOptionPane.showMessageDialog(null, "Enemy left. Returning to the Menu.")
                        client.game.gameFrame.remove(client.game.boardPanel)
                        client.game.createMainMenu()
                    }

                    else -> {}
                }
            } catch (ex: IOException) {
                Logger.getLogger(ClientListenThread::class.java.name).log(Level.SEVERE, null, ex)
            } catch (ex: ClassNotFoundException) {
                println("Girilen class bulunamadı")
            }
        }
    }
}